File: esp/main/checksum.c
#include "embebidos/checksum.h"

#include <stdio.h>
/**
 * @brief Suma cada byte de un mensaje.
 *
 * @param str Puntero a los bytes a los cuales se quiere calcular checksum.
 * @return unsigned int Valor del checksum.
 */
unsigned int checksum(char *str) {
    unsigned int sum = 0;
    while (*str) {
        sum += *str;
        str++;
    }
    return sum;
}

File: cliente/checksum.py
def checksum(msg: bytes) -> int:
    """Calcula el checksum de un mensaje.

    Args:
        msg (bytes): Mensaje a calcular.

    Returns:
        int: Valor del checksum
    """
    sum_value = 0
    for b in msg:
        sum_value += b

    # sum_value += 4294967295
    return sum_value & 0xffffffff
